plugins/modules/consul_agent_check.py

```diff
@@ -0,0 +1,238 @@
+short_description: Add, modify, and delete checks within a consul cluster
+version_added: 8.5.0
+description:
+ - Allows the addition, modification and deletion of checks in a consul
+   cluster via the agent. For more details on using and configuring Checks,
+   see U(https://developer.hashicorp.com/consul/api-docs/agent/check).
+ - Currently, there is no complete way to retrieve the script, interval or TTL
+   metadata for a registered check. Without this metadata it is not possible to
+   tell if the data supplied with ansible represents a change to a check. As a
+   result this does not attempt to determine changes and will always report a
+   changed occurred. An API method is planned to supply this metadata so at that
+   stage change management will be added.
+author:
+  - Michael Ilg (@Ilgmi)
+extends_documentation_fragment:
+  - community.general.consul
+  - community.general.consul.actiongroup_consul
+  - community.general.consul.token
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+    details:
+      - The result is the object as it is defined in the module options and not the object structure of the consul API.
+        For a better overview of what the object structure looks like,
+        take a look at U(https://developer.hashicorp.com/consul/api-docs/agent/check#list-checks).
+  diff_mode:
+    support: partial
+    details:
+      - In check mode the diff will show the object as it is defined in the module options and not the object structure of the consul API.
+options:
+  state:
+    description:
+      - Whether the check should be present or absent.
+    choices: ['present', 'absent']
+    default: present
+    type: str
+  name:
+    description:
+    - Required name for the service check.
+    type: str
+  id:
+    description:
+      - Specifies a unique ID for this check on the node. This defaults to the "Name" parameter, but it may be necessary to provide
+        an ID for uniqueness. This value will return in the response as "CheckId".
+    type: str
+  interval:
+    description:
+      - The interval at which the service check will be run.
+        This is a number with a V(s) or V(m) suffix to signify the units of seconds or minutes, for example V(15s) or V(1m).
+        If no suffix is supplied V(s) will be used by default, for example V(10) will be V(10s).
+      - Required if one of the parameters O(args), O(http), or O(tcp) is specified.
+    type: str
+  notes:
+    description:
+      - Notes to attach to check when registering it.
+    type: str
+  args:
+    description:
+      - Specifies command arguments to run to update the status of the check.
+      - Requires O(interval) to be provided.
+      - Mutually exclusive with O(ttl), O(tcp) and O(http).
+      - "There is an issue with args. It throws an 'Invalid check: TTL must be > 0 for TTL checks'"
+      - See U(https://github.com/hashicorp/consul/issues/6923#issuecomment-564476529) for more details
```

Comment:
What is "it", and what does "throws" mean? This is documentation for an Ansible user, not for a Python programmer.

Also it shouldn't be two paragraphs, but a single one:
```suggestion
      - "There is an issue with args. It throws an 'Invalid check: TTL must be > 0 for TTL checks'.
        See U(https://github.com/hashicorp/consul/issues/6923#issuecomment-564476529) for more details."
```